keras/backend/jax/numpy.py

```diff
@@ -573,7 +578,10 @@ def trace(x, offset=0, axis1=0, axis2=1):
     return jnp.trace(x, offset=offset, axis1=axis1, axis2=axis2)
 
 
-def tri(N, M=None, k=0, dtype="float32"):
+def tri(N, M=None, k=0, dtype=None):
+    # match JAX behavior
+    # https://github.com/google/jax/blob/main/jax/_src/numpy/lax_numpy.py#L2709
+    dtype = dtypes.result_type(dtype or "float32")
```

Comment:
We should default to `floatx()` rather than `"float32"`, no?

Reply:
It is the default behavior from JAX
https://jax.readthedocs.io/en/latest/_modules/jax/_src/numpy/lax_numpy.html#tri

Should we consider changing it?

Reply:
Basically -- yes. Requesting float64 should return a float64 output if the backend supports it (as all do, except JAX).

Reply:
Done.